src/relayFeeCalculator/chain-queries/svmQuery.ts

```diff
@@ -76,24 +83,49 @@ export class SvmQuery implements QueryInterface {
       repaymentAddress
     );
 
-    const [computeUnitsConsumed, gasPriceEstimate] = await Promise.all([
+    const [computeUnitsConsumed, gasPriceEstimate, tokenAccountInfo] = await Promise.all([
       toBN(await this.computeUnitEstimator(fillRelayTx)),
       getGasPriceEstimate(this.provider, {
         unsignedTx: fillRelayTx,
         baseFeeMultiplier: options.baseFeeMultiplier,
         priorityFeeMultiplier: options.priorityFeeMultiplier,
       }),
+      this.provider.getAccountInfo(toAddress(outputToken), { encoding: "base58" }).send(),
     ]);
 
+    // If the owner of the token account is not the token program, then we can assume that it is the 2022 token program address, in which
+    // case we need to determine the extensions the token has to properly calculate rent exemption.
+    const tokenOwner = tokenAccountInfo!.value!.owner;
+    assert(
+      tokenOwner === TOKEN_2022_PROGRAM_ADDRESS || tokenOwner === TOKEN_PROGRAM_ADDRESS,
+      `${outputToken} has invalid token account owner ${tokenOwner}.`
+    );
+    const recipientAta = await getAssociatedTokenAddress(recipient, outputToken, tokenOwner);
+    const encodedAta = await fetchEncodedAccount(this.provider, recipientAta);
+
     // We can cast the gas price estimate to an SvmGasPriceEstimate here since the oracle should always
     // query the Solana adapter.
     const gasPrice = gasPriceEstimate.baseFee.add(
       gasPriceEstimate.microLamportsPerComputeUnit.mul(computeUnitsConsumed).div(toBN(1_000_000)) // 1_000_000 microLamports/lamport.
     );
+    let tokenGasCost = gasPrice;
+
+    // If the ATA does not exist, we need to factor the rent amount into the token gas cost.
+    if (!encodedAta.exists) {
+      // If the ATA is a non-2022 token, then it will always have a fixed size of 165.
+      let extensions = undefined;
+      if (tokenOwner === TOKEN_2022_PROGRAM_ADDRESS) {
+        const mint = await fetchMint(this.provider, toAddress(outputToken));
+        extensions = isSome(mint.data.extensions) ? mint.data.extensions.value : undefined;
+      }
```

Comment:
Are we able to narrow the type of `extensions` ? It's resolving to `any` for me.

Reply:
I believe this does it? https://github.com/across-protocol/sdk/pull/1185/commits/880776eb699a441158ee74773a63e7b4c8bec4f2